lightning-invoice/src/payment.rs

```diff
@@ -1232,14 +1286,36 @@ mod tests {
 	impl lightning::util::ser::Writeable for TestScorer {
 		fn write<W: lightning::util::ser::Writer>(&self, _: &mut W) -> Result<(), std::io::Error> { unreachable!(); }
 	}
+
 	impl Score for TestScorer {
 		fn channel_penalty_msat(
 			&self, _short_channel_id: u64, _send_amt: u64, _chan_amt: Option<u64>, _source: &NodeId, _target: &NodeId
 		) -> u64 { 0 }
 
-		fn payment_path_failed(&mut self, _path: &[&RouteHop], short_channel_id: u64) {
-			if let Some(expected_short_channel_id) = self.expectations.pop_front() {
-				assert_eq!(short_channel_id, expected_short_channel_id);
+		fn payment_path_failed(&mut self, actual_path: &[&RouteHop], actual_short_channel_id: u64) {
+			if let Some(expectation) = self.expectations.pop_front() {
```

Comment:
Hmm, feels like we'd ideally move towards panic'ing if we don't have any expectations here? Fine for now, of course.

Reply:
Good point. Added a panic for when this is called if all expectations have been exhausted.